registration/tests/test_views.py

```diff
@@ -224,14 +224,37 @@ def test_valid_application(self):
         """
         self._assert_registration_succeeds(self.form_data)
 
-    def test_invalid_subdomain(self):
+    @data('hogwarts', 'hog-warts', 'hog1warts')
+    def test_valid_subdomain(self, good_subdomain):
         """
-        Invalid characters in the subdomain.
+        RFC-1034 conformant subdomains succeed in registration.
         """
-        self.form_data['subdomain'] = 'hogwarts?'
+        self.form_data['subdomain'] = good_subdomain
+        self._assert_registration_succeeds(self.form_data)
+
+    @data(
+        '?hogwarts', '?hogwarts?', 'hogwarts?',
+        '-hogwarts', '-hogwarts-', 'hogwarts-',
+        '_hogwarts', 'hog_warts', 'hogwarts_',
+        'hog.warts',
+    )
+    def test_invalid_subdomain(self, bad_subdomain):
+        """
+        Subdomains not RFC-1034 conformant throw an error.
+        """
+        self.form_data['subdomain'] = bad_subdomain
+        self._assert_registration_fails(self.form_data, expected_errors={
+            'subdomain': ['Please include only letters (case-insensitive), numbers, and hyphens. '
+                          'Cannot start or end with a hyphen.'],
+        })
+
+    def test_invalid_long_subdomain(self):
+        """
+        Subdomains not RFC-1034 conformant throw an error.
+        """
+        self.form_data['subdomain'] = 'hogwarts-school-of-witchcraft-and-wizardry-the-un-official-website'
         self._assert_registration_fails(self.form_data, expected_errors={
-            'subdomain': ["Please include only letters, numbers, '_', '-' "
-                          "and '.'"],
+            'subdomain': ['Ensure this value has at most 63 characters (it has 66).'],
         })
```

Comment:
Please also add a tests that illustrates what happens when uniqueness is violated (including case insensitivity).

Reply:
The ``test_existing_subdomain`` already tests for violating uniqueness, and I just realised that with the regex change, it won't be possible to add a subdomain in upper case since it'll already be rejected for violating the regex.